homeassistant/components/zha/sensor.py

```diff
@@ -126,1504 +42,57 @@ async def async_setup_entry(
-    _attribute_name = "measured_value"
-    _attr_device_class: SensorDeviceClass = (
-        SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS_PARTS
-    )
-    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
-    _decimals = 0
-    _multiplier = 1
-    _attr_native_unit_of_measurement = CONCENTRATION_PARTS_PER_BILLION
-
-
-@MULTI_MATCH(cluster_handler_names="pm25")
-# pylint: disable-next=hass-invalid-inheritance # needs fixing
-class PM25(Sensor):
-    """Particulate Matter 2.5 microns or less sensor."""
-
-    _attribute_name = "measured_value"
-    _attr_device_class: SensorDeviceClass = SensorDeviceClass.PM25
-    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
-    _decimals = 0
-    _multiplier = 1
-    _attr_native_unit_of_measurement = CONCENTRATION_MICROGRAMS_PER_CUBIC_METER
-
-
-@MULTI_MATCH(cluster_handler_names="formaldehyde_concentration")
-# pylint: disable-next=hass-invalid-inheritance # needs fixing
-class FormaldehydeConcentration(Sensor):
-    """Formaldehyde Concentration sensor."""
-
-    _attribute_name = "measured_value"
-    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
-    _attr_translation_key: str = "formaldehyde"
-    _decimals = 0
-    _multiplier = 1e6
-    _attr_native_unit_of_measurement = CONCENTRATION_PARTS_PER_MILLION
-
-
-@MULTI_MATCH(
-    cluster_handler_names=CLUSTER_HANDLER_THERMOSTAT,
-    stop_on_match_group=CLUSTER_HANDLER_THERMOSTAT,
-)
-# pylint: disable-next=hass-invalid-inheritance # needs fixing
-class ThermostatHVACAction(Sensor):
-    """Thermostat HVAC action sensor."""
+class Sensor(ZHAEntity, SensorEntity):
+    """ZHA sensor."""
 
-    _unique_id_suffix = "hvac_action"
-    _attr_translation_key: str = "hvac_action"
+    def __init__(self, entity_data: EntityData, **kwargs: Any) -> None:
+        """Initialize the ZHA select entity."""
+        super().__init__(entity_data, **kwargs)
+        entity = self.entity_data.entity
 
-    @classmethod
-    def create_entity(
-        cls,
-        unique_id: str,
-        zha_device: ZHADevice,
-        cluster_handlers: list[ClusterHandler],
-        **kwargs: Any,
-    ) -> Self | None:
-        """Entity Factory.
+        if entity.device_class is not None:
+            self._attr_device_class = SensorDeviceClass(entity.device_class)
 
-        Return entity if it is a supported configuration, otherwise return None
-        """
+        if entity.state_class is not None:
+            self._attr_state_class = SensorStateClass(entity.state_class)
 
-        return cls(unique_id, zha_device, cluster_handlers, **kwargs)
+        if hasattr(entity.info_object, "unit") and entity.info_object.unit is not None:
+            self._attr_native_unit_of_measurement = entity.info_object.unit
 
-    @property
-    def native_value(self) -> str | None:
-        """Return the current HVAC action."""
+        # TODO: In the future, do we migrate all zigpy/zha entities to use entity  # pylint: disable=fixme
```

Comment:
Please do not keep TODOs in source code.

Reply:
removed